cirq-google/cirq_google/named_topologies.py

```diff
@@ -0,0 +1,259 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import dataclasses
+import warnings
+from dataclasses import dataclass
+from functools import lru_cache
+from typing import Dict, List, Tuple, Any, Sequence
+
+import cirq
+import networkx as nx
+from matplotlib import pyplot as plt
+
+
+def cache(user_function):
+    """Unbounded cache.
+
+    Available as functools.cache in Python 3.9+
+    """
+    return lru_cache(maxsize=None)(user_function)
+
+
+def dataclass_json_dict(obj, namespace: str = None):
```

Comment:
nit: ditto for `obj`

(Applies in more places, too.)

Reply:
done